### PR TITLE
Autopark. Level 4. Abstraction

### PR DESCRIPTION
**Pull request by PWMaestro**
Change Engine & CombustionEngine classes on abstract;
Override their ToString() methods;
Create Vehicle method GetVehicleWithMaxMileageOnOneFilling();
Complete level 4 according to the task.

### CODE REVIEW

File: Autopark/Vehicle.cs [hunk 1]
```diff
@@ -80,16 +85,48 @@ public static Vehicle GetVehicleWithMaxMileage()
...
+            }
+            return vehicleWithMaxMileage;
+        }
+
+        public double GetMaxKilometers() => EngineType.GetMaxKilometers(TankVolume);
+
         public double GetCalcTaxPerMonth()
         {   
-            return this.Weight * 0.0013 + Type.TaxCoefficient * 30 + 5;
+            return Weight * 0.0013 + Type.TaxCoefficient * EngineType.TaxCoeffByEngineType * 30 + 5;
```
Comment (dzmitri-litvinov) [inline Autopark/Vehicle.cs:121]: Constant coefficients could be implemented as const private variables in the class

**Review by dzmitri-litvinov — COMMENTED** [review]
Hi:) I have only one comment

**Review by UNDogMan — CHANGES_REQUESTED** [review]
Fix Dmitry comment and it will be ok.

**Review by UNDogMan — APPROVED** [review]
Approve

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**